src/shark/utilityconsole/util/searching/ParameterCriterion.java

```diff
@@ -285,6 +393,70 @@ public int countMatches(ShipHullSpecAPI ship) {
         return retVal;
     }
 
+    public int countMatches(WeaponSpecAPI weapon) {
+        int retVal = 0;
+        switch (this.criteriaParameter.criteria) {
+            case WEAPON: {
+                CriteriaParameter.WeaponCriteriaData actualData = (CriteriaParameter.WeaponCriteriaData) criteriaParameter.criteriaData;
+                // Iterate through all of ship's weapons and make sure they match the criteria
+                if (weapon.getType() == actualData.getType()) {
+                    retVal++;
+                }
+
+                break;
+            }
+            case WEAPON_WITH_SIZE: {
+                CriteriaParameter.WeaponAndSizeCriteriaData actualData = (CriteriaParameter.WeaponAndSizeCriteriaData) criteriaParameter.criteriaData;
+                // Iterate through all of ship's weapons and make sure they match the criteria
+                if (weapon.getType() == actualData.getType() && weapon.getSize() == actualData.getSize()) {
+                    retVal++;
+                }
+
+                break;
+            }
+            case SHIP_PARAMETER:
+                // We simply can't have any of these for weapons
+                // Ex: we can't look for a weapon that has X ballistic slots and Y large missile slots
+                // Ex: we can't look for weapons that satisfy a 'cargo > 500' condition
+                break;
+
+            case WEAPON_PARAMETER: {
+                CriteriaParameter.WeaponParameterCriteriaData actualData = (CriteriaParameter.WeaponParameterCriteriaData) criteriaParameter.criteriaData;
+                switch (actualData.getParameter()) {
+                    case RANGE:
+                        retVal = Math.round(weapon.getMaxRange());
+                        break;
+                    case AMMO:
+                        retVal = weapon.getMaxAmmo();
+                        break;
+                    case TYPE:
+                        retVal = weapon.getType().ordinal(); //TODO what to do here?
+                        break;
+                    case DAMAGE_TYPE:
+                        retVal = weapon.getDamageType().ordinal(); //TODO same thing
+                        break;
+                    case TURN_RATE:
+                        retVal = Math.round(weapon.getTurnRate());
+                        break;
+                    case SIZE:
+                        retVal = weapon.getSize().ordinal(); //TODO similar to hullsize
+                        break;
+                    case RARITY:
+                        retVal = Math.round(weapon.getRarity()); //FIXME i think this is a bug
+                        break;
+                    case BEAM:
+                        retVal = weapon.isBeam() ? 1 : 0;
+                        break;
+                    case MOUNT_TYPE:
+                        retVal = weapon.getMountType().ordinal(); //TODO again
```

Comment:
get rid of these TODOs because there's nothing to do about them. it all worked out.

Reply:
done.

Reply:
resolving.